scene/2d/camera_2d.cpp

```diff
@@ -39,8 +39,10 @@ void Camera2D::_update_scroll() {
 	}
 
 	if (Engine::get_singleton()->is_editor_hint()) {
-		queue_redraw(); //will just be drawn
-		return;
+		queue_redraw();
+		// Only set viewport transform when not bound to the main viewport
+		if (get_viewport() == get_tree()->get_edited_scene_root()->get_viewport())
+			return;
```

Comment:
Two style issues:
- End comment sentences with a period (it's not done consistently in existing code but should be for new code).
- Use brackets even for single line `if` blocks.

Reply:
Fixed!